stock_cycle_count/models/stock_inventory.py

```diff
@@ -90,15 +90,16 @@ def create(self, vals):
 
     def write(self, vals):
         for inventory in self:
-            if (
-                inventory.cycle_count_id
-                and "state" not in vals.keys()
-                and inventory.state == "draft"
-            ):
+            if inventory.not_modify(vals):
                 raise UserError(
                     _(
                         "You cannot modify the configuration of an Inventory "
                         "Adjustment related to a Cycle Count."
                     )
                 )
         return super(StockInventory, self).write(vals)
+
+    def not_modify(self, vals):
+        if self.cycle_count_id and "state" not in vals.keys() and self.state == "draft":
+            return True
+        return False
```

Comment:
simply: `return self.cycle_count_id and "state" not in vals.keys() and self.state == "draft"`